app/src/main/java/nl/tudelft/b_b_w/model/HashException.java

```diff
@@ -3,5 +3,5 @@
 /**
  * Thrown when the hash methods are not available.
  */
-public class HashException extends Exception {
+public class HashException extends RuntimeException {
```

Comment:
If you throw a RuntimeException, the whole app will crash. Why don't you just make your own, and for example in the app it catches it and gives a warning?